src/dbnode/encoding/m3tsz/timestamp_iterator.go

```diff
@@ -260,7 +258,7 @@ func (it *TimestampIterator) readDeltaOfDelta(
 				return 0, nil
 			}
 
-			return xtime.FromNormalizedDuration(dod, timeUnit), nil
+			return xtime.ToUnixNanoDuration(xtime.FromNormalizedDuration(dod, timeUnit)), nil
```

Comment:
Is this necessary? Seems like you're converting to a duration then back to unix nanos

Reply:
I think because `timeUnit` can technically be some unit other than nanoseconds, we have to de-normalize from that unit first, right?

Reply:
Can you just keep the return of this method `readDeltaOfDelta` as `time.Duration`?